event-handler/src/main/java/com/optimizely/ab/android/event_handler/EventDAO.java

```
@@ -120,17 +135,24 @@ boolean removeEvent(long eventId) {
         String selection = EventTable._ID + " = ?";
         // Specify arguments in placeholder order.
         String[] selectionArgs = {String.valueOf(eventId)};
-        // Issue SQL statement.
-        int numRowsDeleted = dbHelper.getWritableDatabase().delete(EventTable.NAME, selection, selectionArgs);
-
-        if (numRowsDeleted > 0) {
-            logger.info("Removed event with id {} from db", eventId);
-            return true;
-        } else {
-            logger.error("Tried to remove an event id {} that does not exist", eventId);
+
+        try {
+            // Issue SQL statement.
+            int numRowsDeleted = dbHelper.getWritableDatabase().delete(EventTable.NAME, selection, selectionArgs);
+
+            if (numRowsDeleted > 0) {
+                logger.info("Removed event with id {} from db", eventId);
+                return true;
+            } else {
+                logger.error("Tried to remove an event id {} that does not exist", eventId);
+            }
+
+            return numRowsDeleted > 0;
+        } catch (Exception e) {
+            logger.error("Could not open db.", e);
```

We are going to just catch all exceptions for db and network I/O since runtime exceptions can be thrown and we don't know which ones they are.